subsys/net/ip/net_tc.c

```diff
@@ -52,6 +52,13 @@ bool net_tc_submit_to_tx_queue(uint8_t tc, struct net_pkt *pkt)
 #if NET_TC_TX_COUNT > 0
 	net_pkt_set_tx_stats_tick(pkt, k_cycle_get_32());
 
+	if (k_sem_take(&tx_classes[tc].fifo_slot, K_NO_WAIT) != 0) {
```

Comment:
Could we hide this behind `#if NET_TC_TX_COUNT > 1`? IMO there's no need to waste cycles on semphore operations that should have no effect if NET_TC_TX_COUNT == 1.

Same comment for the RX part.